### DISCUSSION TITLE
Search in multiple selected drives

### DISCUSSION THREAD

**Discussion by luccpaiva** [kelleyma49/PSFzf#280] [Q&A]
I am trying to come up with a function that would search on multiple drives (C: and D: for me):
```
function Search-CAndD {
    Get-ChildItem C:\, D:\ -Recurse -Force | Select-Object FullName | Invoke-Fzf
}
Set-PSReadLineKeyHandler -Key Ctrl+g -ScriptBlock { Search-CAndD }
```

but this only returns C: path results, even if I call it from the D: root (by cd it first do D:). Besides, this seems slower than the regular command (the command with default bind of 'ctrl + t')

**Comment by msabathier** [comment]
Yes, the `Get-ChildItem` cmdlet is very slow compared to the fzf builtin walker.
You are probably not seeing the D drive file because the `Get-ChildItem` is still walking the C drive which is pretty dense if you have windows installed on it.

Here are a few suggestions to solve this issue : 
- If you can, download and install `fd` and use that rather than the slow `Get-ChildItem`, for example : 
`fd --search-path C:\ --search-path D:\ | fzf`
- If you absolutely want to use the Windows `Get-ChildItem`:
`Get-ChildItem C:\,D:\ -Recurse -Exclude C:\Windows -ErrorAction Ignore | Invoke-Fzf`
  - Add the `-Exclude C:\Windows` option to `Get-ChildItem` to avoid walking the Windows directory (you can also add other directories that you are not interested in searching or even start in C:\Users\luccpaiva if all you want to search is your personal files)
  - Add the `-ErrorAction Ignore` option to ignore errors that might occur during the search (when `Get-ChildItem` tries to open a directory that you don't have access rights for example)
  - Remove the `Select-Object FullName` cmdlet as this will just slow down the search uselessly.
